rumqttd/src/router/iobufs.rs

```diff
@@ -175,6 +179,18 @@ impl Outgoing {
         Some(())
     }
 
+    pub fn register_pubrec(&mut self, pkid: u16) {
+        // NOTE: we can return true of false
+        // to indicate whether this is duplicate or not
+        self.pending_acks.insert(pkid);
+    }
+
+    // PubComp can be out of order!
+    // return false if pkid was not present
```

Comment:
because we don't support out of order acks for other types of acks i would prefer if we keep the behaviour consistent, wdyt?

Reply:
But [standards](https://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718105) don't have any rules for ordering on pubcomp, therefore I thought it implies they can be out or order as well.

Reply:
yes they can be out of order, but [we already don't allow out of order puback's and pubrec's](https://github.com/search?q=repo%3Abytebeamio%2Frumqtt%20ooo&type=code) so to keep things consistent i think we should have same behaviour here as well.

Reply:
I believe we should adhere to standards and let them be out of order, otherwise there is chance someone facing an issue as they though it's okie to send PubComp out of order. Therefore, it seems to be a better choice for long term, wdyt?

Reply:
Anyway, for now, I made the changes to check for order. We will see how it goes later.